runtime/jvmti/jvmtiStartup.c

```diff
@@ -1061,6 +1066,171 @@ createXrunLibraries(J9JavaVM * vm)
+ * @param[in] libNameAr the platform independent agent library name added prefix(lib) and suffix(.a)
+ * @param[in] libpath the library path returned via loadquery(L_GETLIBPATH)
+ *
+ * @return TRUE if the agent library is the same as the object file, FALSE otherwise
+ */
+static BOOLEAN
+searchObjectFileFromLDInfo(const char *fileName, const char *libName, BOOLEAN decorate, BOOLEAN isLibraryPath,
+		const char *libNameSo, const char *libNameAr, char *libpath)
+{
+	BOOLEAN result = FALSE;
+	/* locate the last occurrence of file path separator in the object file */
+	const char *tmp = strrchr(fileName, '/');
+	if (NULL == tmp) {
+		/* this object file has no path separator */
+		if (decorate) {
+			/* compare with libName added the prefix/suffix */
+			if ((0 == strcmp(fileName, libNameSo))
+				|| (0 == strcmp(fileName, libNameAr))
+			) {
+				result = TRUE;
+			}
+		} else {
+			if (!isLibraryPath) {
+				/* libName has no path separator, compare the whole file names */
+				if (0 == strcmp(fileName, libName)) {
+					result = TRUE;
+				}
+			}
+			/* if libName contains a path separator, it can't be the same library as this fileName */
+		}
+	} else {
+		/* find the library name of this object file */
+		const char *tmpLibName = tmp + 1;
+		if (decorate) {
+			/* compare with libName added the prefix/suffix */
+			if ((0 == strcmp(tmpLibName, libNameSo))
+				|| (0 == strcmp(tmpLibName, libNameAr))
+			) {
+				/* check if this object file path is part of libpath */
+				UDATA dirLen = tmp - fileName;
+				result = isFileNameInLibPath(fileName, dirLen, libpath);
+			}
+		} else {
+			/* check if libName contains a path separator */
+			if (isLibraryPath) {
+				/* compare the library directly with the object file loaded */
+				if (0 == strcmp(fileName, libName)) {
+					Trc_JVMTI_searchObjectFileFromLDInfo_library_object_file_found(fileName);
+					result = TRUE;
+				} else {
+					Trc_JVMTI_searchObjectFileFromLDInfo_library_object_file_not_found(fileName, libName);
+				}
+			} else {
+				/* libName has no path separator, compare the whole file names */
+				if (0 == strcmp(tmpLibName, libName)) {
+					/* check if this object file path is part of libpath */
+					UDATA dirLen = tmp - fileName;
+					result = isFileNameInLibPath(fileName, dirLen, libpath);
+				}
+			}
+		}
+	}
+	return result;
+}
+
+/**
+ * A loadquery() helper method to reallocate the buffer if the initial allocation is insufficient.
+ * The caller is responsible for freeing the buffer allocated.
+ *
+ * @param[in] vm Java VM
+ * @param[in] flag specifies the action of the loadquery
+ * @param[in/out] buffer points to a buffer in which to store the information
+ * @param[in/out] bufferLength specifies the initial allocation size and returns the actual size of the buffer
+ *
+ * @return TRUE if loadquery() completes successfully, FALSE otherwise
+ */
+static BOOLEAN
+loadqueryHelper(J9JavaVM *vm, int flag, void **buffer, UDATA *bufferLength)
```

Comment:
This is only used once, with `flag == L_GETINFO`; suggest refactoring the function as
```
static struct ld_info * loadqueryHelper(J9JavaVM *vm)
```
It would return a buffer of `L_GETINFO` data or `NULL` on failure.

Reply:
Changed it to `static struct ld_info * loadqueryGetObjectFiles(J9JavaVM *vm)`.